optuna/pruners/base.py

```diff
@@ -30,3 +31,26 @@ def prune(self, study, trial):
         """
 
         raise NotImplementedError
+
+    @abc.abstractmethod
+    def get_trial_pruner_auxiliary_data(self, study_id, trial_number):
```

Comment:
As `study_id` is deprecated, should this be `study_name` as follows?
```suggestion
    def get_trial_pruner_auxiliary_data(self, study_name, trial_number):
```